array-reduce/main.js

```diff
@@ -0,0 +1,29 @@
+const numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+
+const account = [
+  { type: 'deposit', amount: 150 },
+  { type: 'deposit', amount: 20 },
+  { type: 'withdrawal', amount: 5 },
+  { type: 'deposit', amount: 100 },
+  { type: 'withdrawal', amount: 25 },
+  { type: 'withdrawal', amount: 60 }
+];
+
+const traits = [
+  { color: 'yellow' },
+  { type: 'electric' },
+  { name: 'pikachu' },
+  { level: 15 },
+  { trainer: 'ash' }
+];
+
+const sum = numbers.reduce((init, num) => init + num);
+const product = numbers.reduce((init, num) => init * num);
+const init = 0;
+const balance = account.reduce((previous, current) => {
+  if (current.type === 'deposit') { return previous + current.amount; } else { return previous - current.amount; }
+}, init);
+const composite = traits.reduce((obj, current) => {
+  Object.assign(obj, current);
+  return obj;
+});
```

Comment:
Be sure to pass an initial value (empty object) so that your traits don't all get copied onto the first trait object in the array.